### PR TITLE
[1274][ADD] account_financial_risk_ext tashiro

### PR DESCRIPTION
**Pull request by yostashiro**
Task [1274](https://www.quartile.co/web?debug#id=1274&action=771&model=project.task&view_type=form&menu_id=505)

This is a preceding adjustment to add the credit balance field to res.partner.  Partner list view adjustment to follow in another PR.

### CODE REVIEW

File: account_financial_risk_ext/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2019 Quartile Limited
```
Comment (TimLai125) [inline account_financial_risk_ext/__manifest__.py]: `# Copyright 2020 Quartile Limited`

File: account_financial_risk_ext/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2019 Quartile Limited
```
Comment (TimLai125) [inline account_financial_risk_ext/models/res_partner.py]: `# Copyright 2019 Quartile Limited`

File: account_financial_risk_ext/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    credit_balance = fields.Monetary(compute="_compute_risk_exception")
```
Comment (TimLai125) [inline account_financial_risk_ext/models/res_partner.py]: Should we add `string="Credit Balance"`?

File: account_financial_risk_ext/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    @api.multi
+    @api.depends(lambda x: x._get_depends_compute_risk_exception())
```
Comment (TimLai125) [inline account_financial_risk_ext/models/res_partner.py:13]: Recently I have been wondering what is the purpose of adding `@api.depends` to the compute method of a non-stored compute field.

File: account_financial_risk_ext/views/res_partner_view.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="res_partner_view_risk" model="ir.ui.view">
+        <field name="name">res.partner.view.risk</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="account_financial_risk.res_partner_view_risk" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='credit_limit']" position="after">
+                <field name="credit_balance" widget="monetary" />
```
Comment (TimLai125) [inline account_financial_risk_ext/views/res_partner_view.xml:9]: We should add `attrs="{'readonly': [('risk_allow_edit', '=', False)]}"`, just like other fields in the view
https://github.com/qrtl/asx-custom/blob/2dee897ace548b4df00641c0aee5d94372bf0dca/account_financial_risk/views/res_partner_view.xml#L53-L57

**Review by TimLai125 — CHANGES_REQUESTED** [review]

File: account_financial_risk_ext/models/res_partner.py (hunk 3 above)
Comment (yostashiro) [reply account_financial_risk_ext/models/res_partner.py]: @TimLai125 It is not necessary as the string automatically becomes "Credit Balance", although adding it makes it more explicit.

File: account_financial_risk_ext/views/res_partner_view.xml (hunk 5 above)
Comment (yostashiro) [reply account_financial_risk_ext/views/res_partner_view.xml:9]: @TimLai125 It's a non-stored computed field and, as such, is always readonly.

File: account_financial_risk_ext/models/res_partner.py (hunk 4 above)
Comment (yostashiro) [reply account_financial_risk_ext/models/res_partner.py:13]: @TimLai125 From my observation, it seems that adding `@api.depends` gives you the benefit of computing the field value on the spot when editing the record, whereas, without it, you need to wait until you save the record to trigger the computation.  Can you confirm this behavior when you get the chance?

Comment (TimLai125) [reply account_financial_risk_ext/models/res_partner.py:13]: Let me check it, it could be a nice hint to one of an old SST ticket

**Review by Ryoko04 — APPROVED** [review]

**Review by TimLai125 — APPROVED** [review]
LGTM!

**Pull request merged**